snapcraft/storeapi/errors.py

```diff
@@ -70,3 +73,50 @@ class StoreAuthenticationError(StoreError):
 
     def __init__(self, message):
         super().__init__(message=message)
+
+
+class StoreRegistrationError(StoreError):
+
+    __FMT_ALREADY_REGISTERED = (
+        'The name {snap_name!r} is already taken.\n\n'
+        'We can if needed rename snaps to ensure they match the expectations '
+        'of most users. If you are the publisher most users expect for '
+        '{snap_name!r} then claim the name at {register_claim_url!r}')
+
+    __FMT_RESERVED = (
+        'The name {snap_name!r} is reserved.\n\n'
+        'If you are the publisher most users expect for '
+        '{snap_name!r} then please claim the name at {register_claim_url!r}')
+
+    __FMT_RETRY_WAIT = (
+        'You must wait {retry_after} minutes before trying to register '
+        'your next snap.')
```

Comment:
I would report this in seconds, because according to Celso production requires a wait of 1 minute. So this will say: you must wait 1 minutes.